client-app/src/common/ToolboxLink.js

```diff
@@ -0,0 +1,54 @@
+/*
+ * This file belongs to Hoist, an application development toolkit
+ * developed by Extremely Heavy Industries (www.xh.io | info@xh.io)
+ *
+ * Copyright © 2019 Extremely Heavy Industries Inc.
+ */
+import {Component} from 'react';
+import {HoistComponent, elemFactory, XH} from '@xh/hoist/core';
+import PT from 'prop-types';
+import {a} from '@xh/hoist/cmp/layout';
+
+@HoistComponent
+class ToolboxLink extends Component {
+    
+    static propTypes = {
+        
+        /**
+         * URL for the link, can be a full URL (must include https://) or a string that starts with:
+         *      '$TB' for toolbox files, such as '$TB/client-app/src/desktop/tabs/other/PopupsPanel.js', or
+         *      '$HR' for hoist-react files, such as '$HR/desktop/cmp/button/Button.js'.
+         */
+        url: PT.string.isRequired,
+        
+        /* text to be displayed in the hyperlink.  Defaults to the text after the last `/` */
+        text: PT.string
+    };
+    
+    render() {
+        return a({
+            href: this.createUrl(),
+            item: this.props.text || this.createDefaultText(),
+            target: '_blank'
+        });
+    }
+    
+    createUrl() {
+        const sourceUrls = XH.getConf('sourceUrls');
+        
+        return (this.props.url
+            .replace('$TB', sourceUrls.toolbox)
+            .replace('$HR', sourceUrls.hoistReact)
+        );
+    }
+    
+    createDefaultText() {
```

Comment:
@amcclain 
I took it a step further and wanted to filter out the stuff after `#`, such as when given something like '  I happen to use this a fair bit, so I though it'd be useful.  Open to suggestions :)